registry/api/v2/urls.go

```diff
@@ -174,15 +174,23 @@ func (ub *URLBuilder) BuildBlobUploadURL(name reference.Named, values ...url.Val
 // including any url values. This should generally not be used by clients, as
 // this url is provided by server implementations during the blob upload
 // process.
-func (ub *URLBuilder) BuildBlobUploadChunkURL(name reference.Named, uuid string, values ...url.Values) (string, error) {
+func (ub *URLBuilder) BuildBlobUploadChunkURL(name reference.Named, uuid string, relative bool, values ...url.Values) (string, error) {
 	route := ub.cloneRoute(RouteNameBlobUploadChunk)
 
+	if relative {
+		uploadURL, err := route.relativeURL("name", name.Name(), "uuid", uuid)
+		if err != nil {
+			return "", err
+		}
+		return appendValuesURL(uploadURL, values...).String(), nil
+	}
```

Comment:
As an alternative to splitting the function, could remove some duplicated code by only making the `route.relativeURL` / `route.URL` function calls dependent on `relative`, or even setting a `func` variable to the appropriate function and then having a unified code path from that point.